spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -99,11 +107,24 @@ module.exports = function (provider, setup) {
       });
   });
 
-  it("Pause and Resume work", function (done) {
+  it("Closes socket and open new one on the same port", function (done) {
+    socket.listen('127.0.0.1', 9981, function () {
+      socket.close(function () {
+        socket.listen('127.0.0.1', 9981, function () {
+          expect(true).toEqual(true);
+          PromiseCompat.all([
+            socket.close(function () {}),
+            done()]);
+        });
+      });
+    });
+  });
+
+  it("Pauses and resumes", function (done) {
     socket.connect('www.google.com', 80, function () {
```

Comment:
Looks to me like this test will be inherently flakey... if google sends data back quickly, the data can arrive before you get the chance for pause to become active. Or is the socket provider supposed to also queue stuff synchronously with the set-paused call?